src/PoissonSolvers/FFTOpenPoissonSolver.hpp

```diff
@@ -1141,6 +1203,124 @@ namespace ippl {
             }
             IpplTimings::stopTimer(ifftshift);
 
+        } else if (alg == Algorithm::DCT_VICO) {
+            Vector_t l(hr_m * nr_m);
+            Vector_t hs_m;
+            double L_sum(0.0);
+
+            // compute length of the physical domain
+            // compute Fourier domain spacing
+            for (unsigned int i = 0; i < Dim; ++i) {
+                hs_m[i] = pi * 0.5 / l[i];
+                L_sum   = L_sum + l[i] * l[i];
+            }
+
+            // define the origin of the 2N+1 grid
+            Vector_t origin = 0.0;
+
+            // set mesh for the 2N+1 mesh
+            mesh2n1_m->setMeshSpacing(hs_m);
+
+            // size of truncation window
+            L_sum = std::sqrt(L_sum);
+            L_sum = 1.1 * L_sum;
+
+            // initialize grn2n1_m
+            Vector<int, Dim> size = nr_m;
+
+            // initialize grn2n1_m
+            typename Field_t::view_type view_g2n1 = grn2n1_m.getView();
+            const int nghost_g2n1                 = grn2n1_m.getNghost();
+            const auto& ldom_g2n1                 = layout2n1_m->getLocalNDIndex();
+
+            Kokkos::parallel_for(
+                "Initialize 2N+1 Green's function ", grn2n1_m.getFieldRangePolicy(),
+                KOKKOS_LAMBDA(const int i, const int j, const int k) {
+                    // go from local indices to global
+                    const int ig = i + ldom_g2n1[0].first() - nghost_g2n1;
+                    const int jg = j + ldom_g2n1[1].first() - nghost_g2n1;
+                    const int kg = k + ldom_g2n1[2].first() - nghost_g2n1;
+
+                    double t = ig * hs_m[0];
+                    double u = jg * hs_m[1];
+                    double v = kg * hs_m[2];
+
+                    double s = (t * t) + (u * u) + (v * v);
+                    s        = std::sqrt(s);
+
+                    view_g2n1(i, j, k) =
+                        -2 * (std::sin(0.5 * L_sum * s) / s) * (std::sin(0.5 * L_sum * s) / s);
+
+                    // if (0,0,0), assign L^2/2 (analytical limit of sinc)
+                    if ((ig == 0 && jg == 0 && kg == 0)) {
```

Comment:
The if statement for the origin check should be removed as we want to avoid ifs in Kokkos loops. There should be something like an isOrigin boolean, like it is done in the green's function initialization loops for Hockney and normal Vico.